### PR TITLE
Tweak the search field based on design input

### PR DESCRIPTION
**Pull request by csillag**
- [x] Update the placeholder
- [x] Add missing clear button
- [x] Add time delay between typing and showing the error message
- [x] ~Add animation when showing/hiding search suggestions~
- [x] ~Add animation when showing/hiding the error message~
- [x] Different colors and icons for warning and error messages
- [x] Update borders
- [x] Other updates?
- [x] ~Move some if the customizations to a higher level, into a dedicated (styled) component or into the theme itself~

### CODE REVIEW

File: src/app/components/Search/index.tsx [hunk 1]
```diff
@@ -154,13 +156,26 @@ const SearchCmp: FC<SearchProps> = ({ scope, variant, disabled, onFocusChange: o
...
+  const hasWarning = !!warningMessage
 
-  const hasError = !!errorMessage
+  const hasProblem = hasError || hasWarning
+
+  useEffect(() => {
+    if (hasProblem) {
+      const timeout = setTimeout(() => {
+        setItsTimeToAct(true)
+      }, 500)
```
Comment (lukaw3d) [inline src/app/components/Search/index.tsx:173]: (I dont like this pattern much. I'd validate either instantly, or on submit)

Comment (csillag) [reply src/app/components/Search/index.tsx:173]: Actually we validate instantly in the sense that the search button is disabled right away; we just don't want to shove the error message to the face of the user. (Hopefully he will keep typing and won't even have to see it at all.)

File: src/app/components/Search/index.tsx [hunk 2]
```diff
@@ -105,6 +107,7 @@ const SearchCmp: FC<SearchProps> = ({ scope, variant, disabled, onFocusChange: o
   const [value, setValue] = useState('')
   const [isFocused, setIsFocused] = useState(false)
   const valueInSearchParams = useSearchParams()[0].get('q') ?? ''
+  const [itsTimeToAct, setItsTimeToAct] = useState(false)
```
Comment (lukaw3d) [inline src/app/components/Search/index.tsx:110]: I dont think this variable name is clear

File: src/app/components/Search/index.tsx [hunk 3]
```diff
@@ -154,13 +157,26 @@ const SearchCmp: FC<SearchProps> = ({ scope, variant, disabled, onFocusChange: o
...
+  const hasWarning = !!warningMessage
 
-  const hasError = !!errorMessage
+  const hasProblem = hasError || hasWarning
+
+  useEffect(() => {
+    if (hasProblem) {
+      const timeout = setTimeout(() => {
+        setItsTimeToAct(true)
+      }, tooltipDelay * 2)
```
Comment (lukaw3d) [inline src/app/components/Search/index.tsx:174]: Is there a relation to tooltip?

File: src/app/components/Search/index.tsx (hunk 2 above)
Comment (lukaw3d) [reply src/app/components/Search/index.tsx:110]: Perhaps a better structure is: a delayed "has user stopped typing", instead of delayed "has a problem"

**Review by lukaw3d — APPROVED** [review]

File: src/app/components/Search/index.tsx (hunk 3 above)
Comment (csillag) [reply src/app/components/Search/index.tsx]: No. I have added a separate time constant now.

File: src/app/components/Search/index.tsx (hunk 2 above)
Comment (csillag) [reply src/app/components/Search/index.tsx]: OK, as suggested, now we have `isTyping`, which is always set to true when typing, and then set to false in a delayed manner. (And we display messages when !isTyping.)

**Pull request merged**